### ISSUE TITLE
Add target network to stabilize DQN training

### ISSUE THREAD

**Issue by chirayu08** [chirayu08/dql-traffic-signal#1]
## Problem

The current DQN implementation uses the same network for both
selecting actions and calculating target Q-values.

This can lead to unstable Q-value updates and training loss
oscillation during reinforcement learning.

The README also identifies the absence of a target network as
a current limitation.

## Proposed Solution

Introduce a separate target network for the DQN.

The target network should:

- Start with the same parameters as the policy network.
- Be used to calculate target Q-values.
- Be synchronized with the policy network periodically.
- Keep the existing training pipeline unchanged otherwise.

For example:

    target_net.load_state_dict(policy_net.state_dict())

The target network can be synchronized every N training steps
or episodes.

## Expected Benefits

- More stable Q-value targets
- Reduced training oscillation
- Improved DQN convergence
- Better separation between action selection and target estimation

## Acceptance Criteria

- [ ] Add a target DQN network.
- [ ] Initialize it with the policy network weights.
- [ ] Use the target network when calculating target Q-values.
- [ ] Periodically synchronize target network weights.
- [ ] Verify that the training pipeline still runs correctly.
- [ ] Document the target-network update frequency.
